### PR TITLE
PLANET-6980 Remove restricted tags box

### PR DESCRIPTION
**Pull request by mleray**
### Description

See [PLANET-6980](https://jira.greenpeace.org/browse/PLANET-6980)
This was added to make sure that non-admins cannot create new tags, but that's [now handled in the new sidebar tags selector](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/980). Also, the "normal" tags box was also not properly removed, resulting in 2 tags boxes showing 😬 

### Testing

On local or on the [umbriel test instance](https://www-dev.greenpeace.org/test-umbriel/), whether you login as an admin or as a non-admin, you should see only one "Tags" section in the sidebar. The only difference between the two interfaces should be the "Add new tag" button available for admins, for other users instead they should see the explanation message.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [umbriel](https://www-dev.greenpeace.org/test-umbriel) | [admin](https://www-dev.greenpeace.org/test-umbriel/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-umbriel/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-umbriel) | [composer-local.json](https://github.com/greenpeace/planet4-test-umbriel/blob/main/composer-local.json)

:watch: 2022.12.07 11:03:32

File: src/MasterSite.php [hunk 1]
```diff
@@ -1107,55 +1088,6 @@ public function disallow_insert_term( $term, $taxonomy ) {
...
-				'tags'          => $all_post_tags,
-				'assigned_tags' => $assigned_ids,
-			]
-		);
-	}
-
-	/**
-	 * Remove default WordPress tags selection box for all roles besides administrator.
-	 */
-	public function remove_default_tags_box() {
```
Comment (lithrel) [inline src/MasterSite.php:1136]: this one is still called on l.142 ?

Comment (mleray) [reply src/MasterSite.php:1136]: Good catch 😬 PR updated!

**Review by lithrel — APPROVED** [review]
:+1:

**Pull request merged**